Tools/AP_Periph/can.cpp

```diff
@@ -1298,16 +1367,15 @@ void AP_Periph_FW::can_start()
         if (can_iface_periph[i] != nullptr) {
             can_iface_periph[i]->init(g.can_baudrate[i], AP_HAL::CANIface::NormalMode);
         }
-    }
-
-    canardInit(&canard, (uint8_t *)canard_memory_pool, sizeof(canard_memory_pool),
-               onTransferReceived, shouldAcceptTransfer, NULL);
+        canardInit(&canard[i], (uint8_t *)canard_memory_pool[i], sizeof(canard_memory_pool[i]),
+                onTransferReceived, shouldAcceptTransfer, NULL);
 
-    if (PreferredNodeID != CANARD_BROADCAST_NODE_ID) {
-        canardSetLocalNodeID(&canard, PreferredNodeID);
+        if (PreferredNodeID != CANARD_BROADCAST_NODE_ID) {
+            canardSetLocalNodeID(&canard[i], PreferredNodeID);
+        }
     }
 
-    // wait for dynamic node ID allocation
+    // wait for dynamic node ID allocation on atleast one iface
```

Comment:
"at least" is two words

Reply:
What is the consequence of an iface running without an allocated Node ID yet?

Reply:
we just wait on allocation on that iface, until anything is transmitted. the allocated one goes on as usual